it/test/controllers/agentAppointment/AppointAgentsControllerISpec.scala

```diff
@@ -776,18 +781,287 @@ class AppointAgentsControllerISpec extends ISpecBase with HtmlComponentHelpers {
+        document.select(errorLocator).text shouldBe selectWhichPropertiesErrorTextWelsh
+      }
+
+      s"has correct post url" in {
+        val secondForm = document.select("form").get(1) // Nested form
+        val actionUrl = secondForm.attr("action")
+        val expectedUrl =
+          "/business-rates-property-linking/my-organisation/appoint/properties?page=1&pageSize=15&agentCode=1001&backLinkUrl=%2Fbusiness-rates-property-linking%2Fmy-organisation%2Fappoint%2Fproperties%3Fpage%3D1%26pageSize%3D15%26agentCode%3D1001%26backLinkUrl%3Dhttp%253A%252F%252Flocalhost%252Fsome-back-link%26fromManageAgentJourney%3Dfalse&fromManageAgentJourney=false"
+
+        actionUrl shouldBe expectedUrl
+      }
+    }
+
+  }
+
+  def postSearchBadRequest(language: Language, withAgents: Boolean = false, manageAgentJourney: Boolean) = {
+
+    val testSessionId = s"stubbed-${UUID.randomUUID}"
+    val agentsCode = 1001L
+    val agentName = "Test Agent"
+    val backLinkUrl = RedirectUrl("http://localhost/some-back-link")
+
+    commonStubs(testSessionId, agentsCode, agentName, withAgents)
+
+    val requestBody = Json.obj(
+    )
+
+    await(
+      ws.url(
+        s"http://localhost:$port/business-rates-property-linking/my-organisation/appoint/properties/filter?page=1&pageSize=15&agentCode=$agentsCode&backLinkUrl=${backLinkUrl.unsafeValue}&fromManageAgentJourney=$manageAgentJourney"
+      ).withCookies(languageCookie(language), getSessionCookie(testSessionId))
+        .withFollowRedirects(follow = false)
+        .withHttpHeaders(HeaderNames.COOKIE -> "sessionId", "Csrf-Token" -> "nocheck")
+        .post(requestBody)
+    )
+  }
+
+  def postBadRequest(language: Language, withAgents: Boolean = false, manageAgentJourney: Boolean) = {
+
+    val testSessionId = s"stubbed-${UUID.randomUUID}"
+    val agentsCode = 1001L
+    val agentName = "Test Agent"
+    val backLinkUrl = RedirectUrl("http://localhost/some-back-link")
+
+    commonStubs(testSessionId, agentsCode, agentName, withAgents)
+
+    val requestBody = Json.obj(
+      "agentCode"   -> agentsCode,
+      "name"        -> agentName,
+      "backLinkUrl" -> s"${backLinkUrl.unsafeValue}"
+    )
+
+    await(
+      ws.url(
+        s"http://localhost:$port/business-rates-property-linking/my-organisation/appoint/properties?agentCode=$agentsCode&backLinkUrl=${backLinkUrl.unsafeValue}&fromManageAgentJourney=$manageAgentJourney"
+      ).withCookies(languageCookie(language), getSessionCookie(testSessionId))
+        .withFollowRedirects(follow = false)
+        .withHttpHeaders(HeaderNames.COOKIE -> "sessionId", "Csrf-Token" -> "nocheck")
+        .post(requestBody)
+    )
   }
 
   def getMyOrganisationPropertyLinksWithAgentFilteringPage(
         language: Language,
         withAgents: Boolean = false,
-        withAgentsNotAppointed: Boolean = false
+        withAgentsNotAppointed: Boolean = false,
+        badRequest: Boolean = false,
+        manageAgentJourney: Boolean = false
```

Comment:
I can't see where you're using `badRequest` or `manageAgentJourney`. This might be left over before you split it out into `postBadRequest` and `postSearchBadRequest`